src/components/application_manager/src/commands/command_request_impl.cc

```diff
@@ -605,11 +627,128 @@ bool CommandRequestImpl::HasDisallowedParams() const {
+    first.interface_state = application_manager_.hmi_interfaces().GetInterfaceState(
+            first.interface);
+    second.interface_state = application_manager_.hmi_interfaces().GetInterfaceState(
+            second.interface);
+
+    bool result = (first.is_ok && second.is_ok) ||
+                  (second.is_invalid_enum && first.is_ok) ||
+                  (first. is_invalid_enum && second.is_ok);
+
+    result = result || (first.is_ok && second.is_unsupported_resource);
+    result = result || (second.is_ok && first.is_unsupported_resource);
+    result = result || (first.is_invalid_enum && second.is_unsupported_resource &&
+                        HmiInterfaces::STATE_NOT_AVAILABLE != second.interface_state);
+    result = result || (second.is_invalid_enum && first.is_unsupported_resource &&
+                        HmiInterfaces::STATE_NOT_AVAILABLE != first.interface_state);
+    return result;
+}
+
+void CommandRequestImpl::GetInfo(HmiInterfaces::InterfaceID interface,
+                hmi_apis::Common_Result::eType result_code,
+                std::string& info,
+                const smart_objects::SmartObject& response_from_hmi) {
+    if (response_from_hmi[strings::msg_params].keyExists(strings::info)) {
+        info =
+        response_from_hmi[strings::msg_params][strings::info].asString();
+    } else {
+        HmiInterfaces::InterfaceState interface_state =
+                application_manager_.hmi_interfaces().GetInterfaceState(interface);
+        if (hmi_apis::Common_Result::UNSUPPORTED_RESOURCE == result_code &&
+            HmiInterfaces::InterfaceState::STATE_NOT_RESPONSE == interface_state) {
+            CreateInfoForUnsupportedResult(info, interface);
+        }
+    }
+
+ }
+
+mobile_apis::Result::eType CommandRequestImpl::PrepareResultCodeForResponse(
+    const response_info& first,
+    const response_info& second) {
+    mobile_apis::Result::eType result_code = mobile_apis::Result::INVALID_ENUM;
+    if (((first.is_ok || first.is_invalid_enum) &&
+         second.is_unsupported_resource) ||
+        ((second.is_ok || second.is_invalid_enum) &&
+         first.is_unsupported_resource) ||
+         (first.is_unsupported_resource && second.is_unsupported_resource)) {
+      result_code = mobile_apis::Result::UNSUPPORTED_RESOURCE;
+    } else {
+        //If esponse contains erroneous result code SDL need return erroneus result code.
+        hmi_apis::Common_Result::eType first_result =
+                hmi_apis::Common_Result::INVALID_ENUM;
+        hmi_apis::Common_Result::eType second_result =
+                hmi_apis::Common_Result::INVALID_ENUM;
+        if(!first.is_unsupported_resource) {
+            first_result = first.result_code;
+        }
+        if (!second.is_unsupported_resource) {
+            second_result  = second.result_code;
+        }
+        result_code = MessageHelper::HMIToMobileResult(
+                    std::max(first_result, second_result));
+    }
+    return result_code;
```

Comment:
I think we can at least decompose this method to preparing one argument result and accumulator for both infos.